File: app/src/main/java/io_developers/sssemil/com/wand/Account/ApiHelper.java
package io_developers.sssemil.com.wand.Account;

import android.content.SharedPreferences;
import retrofit.RequestInterceptor;
import retrofit.RestAdapter;
import retrofit.http.Field;
import retrofit.http.FormUrlEncoded;
import retrofit.http.POST;
import rx.Observable;

/**
 * Created by emil on 12/05/17.
 */
public class ApiHelper {
    public static final String PREF_TOKEN = "pref_token";
    public static final String PREF_EMAIL = "pref_email";
    public static final String PREF_NAME = "pref_name";

    private static final String SERVER_URL = "http://wand.ddns.net/";

    private Api mApi;

    public ApiHelper() {
        RequestInterceptor requestInterceptor = new RequestInterceptor() {
            @Override
            public void intercept(RequestFacade request) {
                request.addHeader("Accept", "application/json");
            }
        };

        RestAdapter restAdapter = new RestAdapter.Builder()
                .setEndpoint(SERVER_URL)
                .setRequestInterceptor(requestInterceptor)
                .setLogLevel(RestAdapter.LogLevel.FULL)
                .build();

        mApi = restAdapter.create(Api.class);
    }

    public Api getApi() {
        return mApi;
    }

    public interface Api {
        @POST("/init.php")
        @FormUrlEncoded
        Observable<String> signup(@Field("email") String email, @Field("password") String password,
                                  @Field("name") String name, @Field("mobile") String mobile);

        @POST("/login.php")
        @FormUrlEncoded
        Observable<String> login(@Field("email") String email, @Field("password") String password);
    }

    public static void logOut(SharedPreferences sharedPreferences){
        sharedPreferences.edit().remove(PREF_EMAIL).remove(PREF_NAME).remove(PREF_TOKEN).apply();
    }
}

File: app/src/main/java/iodevelopers/sssemil/com/wand/Account/ApiHelper.kt
package iodevelopers.sssemil.com.wand.Account

import android.content.SharedPreferences
import retrofit.RestAdapter
import retrofit.http.Field
import retrofit.http.FormUrlEncoded
import retrofit.http.POST
import rx.Observable

/**
 * Created by emil on 12/05/17.
 */
class ApiHelper {

    val api: Api

    init {
        val requestInterceptor = retrofit.RequestInterceptor {
            request -> request.addHeader("Accept", "application/json")
        }

        val restAdapter = RestAdapter.Builder()
                .setEndpoint(SERVER_URL)
                .setRequestInterceptor(requestInterceptor)
                .setLogLevel(RestAdapter.LogLevel.FULL)
                .build()

        api = restAdapter.create(Api::class.java)
    }

    interface Api {
        @POST("/init.php")
        @FormUrlEncoded
        fun signup(@Field("email") email: String, @Field("password") password: String,
                   @Field("name") name: String, @Field("mobile") mobile: String): Observable<String>

        @POST("/login.php")
        @FormUrlEncoded
        fun login(@Field("email") email: String, @Field("password") password: String): Observable<String>
    }

    companion object {
        val PREF_TOKEN = "pref_token"
        val PREF_EMAIL = "pref_email"
        val PREF_NAME = "pref_name"

        private val SERVER_URL = "http://wand.ddns.net/"

        fun logOut(sharedPreferences: SharedPreferences?) {
            sharedPreferences!!.edit()
                    .remove(ApiHelper.Companion.PREF_EMAIL)
                    .remove(ApiHelper.Companion.PREF_NAME)
                    .remove(ApiHelper.Companion.PREF_TOKEN).apply()
        }
    }
}